packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-self-reg/java/integrationsteps.md

```diff
@@ -1,3 +1,251 @@
+
+```java
+VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(code);
+AuthenticationResponse authenticationResponse =
+   idxAuthenticationWrapper.verifyAuthenticator(proceedContext, verifyAuthenticatorOptions);
+```
+
+If the request to verify the code is successful, the SDK returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` and an `Authenticators` list containing the phone factor. Reuse the authenticator enrollment form from [Step 3: Display enrollment factors](#step-3-display-enrollment-factors) to display the list of authenticators to the user.
+
+```java
+public ModelAndView selectAuthenticatorForm(AuthenticationResponse response, String title, HttpSession session) {
+   boolean canSkip = authenticationWrapper.isSkipAuthenticatorPresent(response.getProceedContext());
+   ModelAndView modelAndView = new ModelAndView("select-authenticator");
+   modelAndView.addObject("canSkip", canSkip);
+   List<String> factorMethods = new ArrayList<>();
+   for (Authenticator authenticator : response.getAuthenticators()) {
+      for (Authenticator.Factor factor : authenticator.getFactors()) {
+            factorMethods.add(factor.getMethod());
+      }
+   }
+   session.setAttribute("authenticators", response.getAuthenticators());
+   modelAndView.addObject("factorList", factorMethods);
+   modelAndView.addObject("authenticators", response.getAuthenticators());
+   modelAndView.addObject("title", title);
+   return modelAndView;
+}
+```
+
+Based on the configuration described in [Set up your Okta org for multifactor use cases](/docs/guides/oie-embedded-common-org-setup/java/main/#set-up-your-okta-org-for-multifactor-use-cases), the app in this use case is set up to require one possession factor (either email or phone). After the email factor is verified, the phone factor becomes optional. In this step, the `isSkipAuthenticatorPresent` function returns `TRUE` for the phone authenticator. You can build a **Skip** button in your form to allow the user to skip the optional phone factor. If the user decides to skip the optional factor, they are considered signed in since they have already verified the required factors. See [Step 8b, Option 2: Skip phone factor](#step-8b-option-2-skip-phone-factor) for the skip authenticator flow.
+
+<div class="common-image-format">
+
+![Displays the Java SDK's phone authenticator enrollment form with Skip button](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-auth-list-phone-java.png)
+
+</div>
+
+### Step 8, Option 1: User selects phone authenticator
+
+In this use case option, the user selects the optional **Phone** factor as the authenticator to verify. Pass this selected authenticator to the `IDXAuthenticationWrapper.selectAuthenticator` method.
+
+```java
```

Comment:
Same comment as before: When the password was selected for verification, it was passed an object called `verifyAuthenticatorOptions`:

```
VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(newPassword);
```

Shouldn't this instance use the same preceding object creation line and variable name for consistency, except with `newPhone` instead of `newPassword`?

Reply:
@vanngo-okta ?

Reply:
Not sure if I understand your comment. Maybe we can chat with references to the updated lines in the file?